File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int tc = 10;

    for (int t = 1; t < tc + 1; t++) {
        int n;
        cin >> n;

        vector<int> building(n);
        for (int i = 0; i < n; i++) {
            cin >> building[i];
        }

        int answer = 0;

        for (int i = 2; i < n - 2; ++i) {
            if ((building[i - 2] < building[i] && building[i - 1] < building[i]) &&
                (building[i + 1] < building[i] && building[i + 2] < building[i])) {
                answer += building[i] - max({building[i - 2], building[i - 1], building[i + 1], building[i + 2]});
            }
        }
        
        cout << "#" << t << " " << answer << endl;
    }
    
    return 0;
}

File: SWEA/D3/1206. ［S／W 문제해결 기본］ 1일차 － View/［S／W 문제해결 기본］ 1일차 － View.py
for tc in range(1, 11):
    answer = 0
    n = int(input())
    b = list(map(int, input().split()))
    for i in range(2, n - 2):
        if max(b[i-2], b[i-1], b[i], b[i+1], b[i+2]) == b[i]:
            answer += b[i] - max(b[i-2], b[i-1], b[i+1], b[i+2])

    print(f"#{tc} {answer}")